chain/ethhashlookup/eth_transaction_hash_lookup.go

```diff
@@ -0,0 +1,166 @@
+package ethhashlookup
+
+import (
+	"database/sql"
+	"math"
+
+	"github.com/ipfs/go-cid"
+	_ "github.com/mattn/go-sqlite3"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+)
+
+var pragmas = []string{
+	"PRAGMA synchronous = normal",
+	"PRAGMA temp_store = memory",
+	"PRAGMA mmap_size = 30000000000",
+	"PRAGMA page_size = 32768",
+	"PRAGMA auto_vacuum = NONE",
+	"PRAGMA automatic_index = OFF",
+	"PRAGMA journal_mode = WAL",
+	"PRAGMA read_uncommitted = ON",
+}
+
+var ddls = []string{
+	`CREATE TABLE IF NOT EXISTS tx_hash_lookup (
+		hash TEXT PRIMARY KEY UNIQUE,
+		cid TEXT NOT NULL UNIQUE,
+		epoch INT NOT NULL
+	)`,
+
+	`CREATE UNIQUE INDEX IF NOT EXISTS tx_hash_lookup_hash_uindex ON tx_hash_lookup (hash)`,
+
+	`CREATE INDEX IF NOT EXISTS tx_hash_lookup_epoch_index ON tx_hash_lookup (epoch)`,
+
+	// metadata containing version of schema
+	`CREATE TABLE IF NOT EXISTS _meta (
+    	version UINT64 NOT NULL UNIQUE
+	)`,
+
+	// version 1.
+	`INSERT OR IGNORE INTO _meta (version) VALUES (1)`,
+}
+
+const schemaVersion = 1
+const MemPoolEpoch = math.MaxInt64
+
+const (
+	insertTxHash = `INSERT INTO tx_hash_lookup
+	(hash, cid, epoch)
+	VALUES(?, ?, ?)
+	ON CONFLICT (hash) DO UPDATE SET epoch = EXCLUDED.epoch
+	WHERE epoch > EXCLUDED.epoch`
+)
+
+type TransactionHashLookup struct {
```

Comment:
General go idiom is to leverage the package name to convey information, instead of duplicating it in type names, e.g. `ethhashlookup.Table` or `ethhash.LookupTable` (latter probably probably better since it defines an area and not just one mechanism).